app/models/ext_management_system.rb

```diff
@@ -101,8 +105,8 @@ def self.supported_for_create?
   validates :name,     :presence => true, :uniqueness => {:scope => [:tenant_id]}
   validates :hostname, :presence => true, :if => :hostname_required?
   validate :hostname_uniqueness_valid?, :hostname_format_valid?, :if => :hostname_required?
-
   validate :validate_ems_enabled_when_zone_changed?, :validate_zone_not_maintenance_when_ems_enabled?
+  validate :validate_ems_type, :on => :create
```

Comment:
Can you add a couple of tests for this validation?

Reply:
Was this resolved?  I see tests on create with valid classes but not invalid classes, probably because the factories will default to using a leaf class.

Reply:
Followup question is [here](https://github.com/ManageIQ/manageiq/pull/18842#discussion_r297808806)